packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1569,6 +1571,86 @@ export class AccountHandler {
 
     return {};
   }
+
+  async loginThirdParty(request: AuthRequest) {
+    const requestPayload = request.payload as any;
+    const token = JSON.parse(requestPayload.token);
+
+    const idToken = token.credential;
+    const CLIENT_ID =
+      '218517873053-th4taguk9dvf03rrgk8sigon84oigf5l.apps.googleusercontent.com';
+    const client = new OAuth2Client(CLIENT_ID);
+    try {
+      const ticket = await client.verifyIdToken({
+        idToken,
+        audience: CLIENT_ID, // Specify the CLIENT_ID of the app that accesses the backend
+        // Or, if multiple clients access the backend:
+        //[CLIENT_ID_1, CLIENT_ID_2, CLIENT_ID_3]
+      });
+      const payload = ticket.getPayload();
+      const userid = payload['sub'];
+      // If request specified a G Suite domain:
+      // const domain = payload['hd'];
+
+      let accountRecord = await this.db.getAccountByGoogleId(userid);
+
+      // No account is associated with this google user
+      if (!accountRecord) {
+        const email = payload.email;
+
+        try {
+          accountRecord = await this.db.accountRecord(payload.email);
+        } catch (err) {
+          if (err.errno === 102) {
+            const { hex16: emailCode, hex32: authSalt } =
+              await this.generateRandomValues();
+            const [kA, wrapWrapKb] = await random.hex(32, 32);
+            accountRecord = await this.db.createAccount({
```

Comment:
Why not update this call to include the google info to avoid the other 2 async calls?